installer/pkg/nextgen/asset/platform.go

```diff
@@ -0,0 +1,44 @@
+package asset
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/openshift/installer/installer/pkg/nextgen"
+)
+
+const (
+	AWSPlatformType     = "aws"
+	LibvirtPlatformType = "libvirt"
+)
+
+type Platform struct{}
```

Comment:
I don't think it much matters whether it is one type or two separate types. In the end, I decided on having Platform as a separate type because there was extra code that needed to go somewhere to define the Validate for the platform and to define the acceptable platforms. Either I put this extra code in stock.go, which will tend to make that file grow too large, or I create a separate file for the extra platform code. If I create a separate file, then I may as well create a separate type, too, which has the extra benefit of keeping the UserProvided type smaller.